### PR TITLE
Set Model position attribute to the 0th index

### PR DESCRIPTION
**Pull request by lilleyse**
This fixes a bug with shadow-casting on linux where the normal attribute is the first attribute in the glTF file but is compiled out of the shadow-cast shader, and so no attributes are bound to the 0th index causing the model shadows to be broken. This fix forces the position attribute to be the 0th index. It's a little weird though, because the `program` node only  lists attribute names and not semantics, so I had to search through the techniques to find the attribute name that is tied to the POSITION semantic.

### CODE REVIEW

**Comment by pjcozzi** [comment]
I'm not sure about this approach.  There's two ways I think we could make it simpler.
* Instead of looking at the semantic, just look at the name with a case insensitive search for `position`.  In practice, it will be reliable and is much more clean.
* Or, as a preprocess (at load time), walk the techniques, find the `POSITION` semantic and then augment the `program` object (then `delete` this after load) with the attribute index.

**Comment by lilleyse** [comment]
Updated using the first approach.

**Comment by pjcozzi** [comment]
Sorry, late request: add a comment about why this is happening and update CHANGES.md if needed.

File: Source/Scene/Model.js [hunk 1]
```diff
@@ -1493,11 +1493,21 @@ define([
...
+    function createAttributeLocations(model, attributes) {
         var attributeLocations = {};
         var length = attributes.length;
+        var i;
 
-        for (var i = 0; i < length; ++i) {
+        // Set the position attribute to the 0th index
+        for (i = 1; i < length; ++i) {
+            var attribute = attributes[i];
+            if (attribute.toLowerCase().indexOf('position') > -1) {
```
Comment (mramato) [inline Source/Scene/Model.js]: Use `/position/i.test(attribute)` here instead, it will be cleaner and faster.

**Comment by lilleyse** [comment]
Updated.

**Comment by pjcozzi** [comment]
Looks good!

**Pull request merged**